lib/rdoc/code_object/constant.rb

```diff
@@ -174,6 +174,20 @@ def store=(store)
     @file = @store.add_file @file.full_name if @file
   end
 
+  ##
+  # Process comment as YARD comment
+  def comment=(comment)
```

Comment:
Handling YARD in `comment=` seems too aggressive. It can be triggered from many different situation and places.
```ruby
def add_alias
  # Maybe this is ok, but I can't confidently say that this is safe
  method.comment = an_alias.comment
end
```

How about explicitly processing it in RDoc::Parser::PrismRuby?
```ruby
def add_comment
  ...
  comment.normalize
  override_visibility, some_other_information = handle_yard(meth, comment)
  if override_visibility
    do_something
  end
  meth.comment = comment
  ...
end
```

`RDoc::Parser::Ruby` is too complicated (and buggy) to maintain, so the system design of YARD-support shouldn't be affected by RDoc::Parser::Ruby implementation detail. I think supporting YARD only in RDoc::Parser::PrismRuby is enough.